rest_framework/serializers.py

```diff
@@ -314,6 +315,17 @@ def fields(self):
                 self._fields[key] = value
         return self._fields
 
+    @cached_property
+    def _to_internal_value_fields(self):
+        return [
+            field for field in self.fields.values()
+            if (not field.read_only) or (field.default is not empty)
+        ]
+
+    @cached_property
+    def _to_representation_fields(self):
+        return [field for field in self.fields.values() if not field.write_only]
```

Comment:
We could consider using the same multiline form as in `_to_internal_value_fields()`, so...

    return [
        field for field in self.fields.values()
        if not field.write_only
    ]

I find that quite readable. ?